### ISSUE TITLE
Terraform provider not finding organisation 

### ISSUE THREAD

**Issue by coetzerstefan** [terraform-google-modules/terraform-google-group#65]
### TL;DR

Using the terraform-google-modules provider to manage GWS groups it's not able to find my GCP/GWS organisation, however, I specified it in the domain attribute. I'm not seeing any open issues online (here/stack overflow) about this apologies if it's something on my end.
Authenticating via service account with the relevant permissions assigned.

### Expected behavior

Terraform should create & manage GWS groups

### Observed behavior

│ Error: Organization not found: testdomain.com
│ 
│   with module.group.data.google_organization.org[0],
│   on .terraform/modules/group/main.tf line 17, in data "google_organization" "org":
│   17: data "google_organization" "org" {
│

### Terraform Configuration

```hcl
module "group" {
  source  = "terraform-google-modules/group/google"
  version = "~> 0.6"
  
  id           = var.group_id
  display_name = var.name
  description  = "Test Google Group via Terraform"
  domain       = "testdomain.com"
  owners       = var.owners
  managers     = var.managers
  members      = var.members
}
```

### Terraform Version

```sh
v1.5.7
```
